proto/redpanda/core/admin/cluster_link.proto

```diff
@@ -0,0 +1,416 @@
+        };
+    }
+    rpc ListClusterLinks(ListClusterLinksRequest)
+        returns (ListClusterLinksResponse) {
+        option (redpanda.pbgen.rpc) = {
+            authz: SUPERUSER
+        };
+    }
+    rpc UpdateClusterLink(UpdateClusterLinkRequest) returns (ClusterLink) {
+        option (redpanda.pbgen.rpc) = {
+            authz: SUPERUSER
+        };
+    }
+    rpc FailOverClusterLink(FailOverClusterLinkRequest) returns (ClusterLink) {
+        option (redpanda.pbgen.rpc) = {
+            authz: SUPERUSER
+        };
+    }
+}
+
+/* Resources */
+/// A ClusterLink resource
+message ClusterLink {
+    /// The name of the cluster link
+    string name = 1 [(google.api.field_behavior) = REQUIRED];
+    /// The UUID of the cluster link
+    string uid = 2 [
+        (google.api.field_behavior) = OUTPUT_ONLY,
+        (google.api.field_info).format = UUID4
+    ];
+    /// Cluster link configuration
+    ClusterLinkConfiguration configuration = 3;
+    /// Status of the cluster link
+    ClusterLinkStatus status = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+/* RPC Requests and Responses */
+/// Create a new cluster link
+message CreateClusterLinkRequest {
+    /// The cluster link to create
+    ClusterLink cluster_link = 1;
+}
+/// Request to delete a cluster link
+message DeleteClusterLinkRequest {
+    /// The name of the link to delete
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (google.api.resource_reference) = {
+            type: "redpanda.core.admin.ClusterLinkService/ClusterLink"
+        }
+    ];
+}
+message DeleteClusterLinkResponse {}
+message GetClusterLinkRequest {
+    /// The name of the cluster link to get
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (google.api.resource_reference) = {
+            type: "redpanda.core.admin.ClusterLinkService/ClusterLink"
+        }
+    ];
+}
+/// Request to list all cluster links
+message ListClusterLinksRequest {}
+/// All cluster links on the cluster
+message ListClusterLinksResponse {
+    /// The cluster links
+    repeated ClusterLink cluster_links = 1;
+}
+/// Updates a cluster link
+message UpdateClusterLinkRequest {
+    /// The cluster link to update
+    ClusterLink cluster_link = 1;
+
+    /// The list of fields to update
+    optional google.protobuf.FieldMask update_mask = 2;
+}
+/// Fails over the cluster link
+message FailOverClusterLinkRequest {
+    /// The name of the cluster link to fail over
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (google.api.resource_reference) = {
+            type: "redpanda.core.admin.ClusterLinkService/ClusterLink"
+        }
+    ];
+}
+
+/* Other Messages */
+/// ClusterLink options
+message ClusterLinkConfiguration {
+    /// Configuration for the internal kafka client
+    ClusterLinkClientOptions client_options = 1;
+    /// Topic metadata sync options
+    /// If not provided, defaults to enabled with no filters
```

Comment:
`defaults to enabled with no filters` reads a bit funny to me. Maybe: `by default all data is fetched from the source cluster`.